File: app/src/test/java/org/schabi/newpipe/report/ErrorActivityTest.java
package org.schabi.newpipe.report;

import android.app.Activity;

import org.junit.Test;
import org.schabi.newpipe.MainActivity;
import org.schabi.newpipe.RouterActivity;
import org.schabi.newpipe.fragments.detail.VideoDetailFragment;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

/**
 * Unit tests for {@link ErrorActivity}
 */
public class ErrorActivityTest {
    @Test
    public void getReturnActivity() {
        Class<? extends Activity> returnActivity;
        returnActivity = ErrorActivity.getReturnActivity(MainActivity.class);
        assertEquals(MainActivity.class, returnActivity);

        returnActivity = ErrorActivity.getReturnActivity(RouterActivity.class);
        assertEquals(RouterActivity.class, returnActivity);

        returnActivity = ErrorActivity.getReturnActivity(null);
        assertNull(returnActivity);

        returnActivity = ErrorActivity.getReturnActivity(Integer.class);
        assertEquals(MainActivity.class, returnActivity);

        returnActivity = ErrorActivity.getReturnActivity(VideoDetailFragment.class);
        assertEquals(MainActivity.class, returnActivity);
    }



}

File: app/src/test/java/org/schabi/newpipe/error/ErrorActivityTest.kt
package org.schabi.newpipe.error

import android.app.Activity
import org.junit.Assert
import org.junit.Test
import org.schabi.newpipe.MainActivity
import org.schabi.newpipe.RouterActivity
import org.schabi.newpipe.error.ErrorActivity.Companion.getReturnActivity
import org.schabi.newpipe.fragments.detail.VideoDetailFragment

/**
 * Unit tests for [ErrorActivity].
 */
class ErrorActivityTest {
    @get:Test
    val returnActivity: Unit
        get() {
            var returnActivity: Class<out Activity?>?
            returnActivity = getReturnActivity(MainActivity::class.java)
            Assert.assertEquals(MainActivity::class.java, returnActivity)
            returnActivity = getReturnActivity(RouterActivity::class.java)
            Assert.assertEquals(RouterActivity::class.java, returnActivity)
            returnActivity = getReturnActivity(null)
            Assert.assertNull(returnActivity)
            returnActivity = getReturnActivity(Int::class.java)
            Assert.assertEquals(MainActivity::class.java, returnActivity)
            returnActivity = getReturnActivity(VideoDetailFragment::class.java)
            Assert.assertEquals(MainActivity::class.java, returnActivity)
        }
}
